libs/utils/src/Card/web_service_card.test.ts

```diff
@@ -103,6 +127,82 @@ it('writes binary data using /card/write_long_b64', async () => {
   await new WebServiceCard().writeLongUint8Array(Uint8Array.of(1, 2, 3));
 });
 
+it('handles failures to write binary data using /card/write_long_b64', async () => {
+  fetchMock.post('/card/write_long_b64', { success: false });
+
+  await expect(
+    new WebServiceCard().writeLongUint8Array(Uint8Array.of(1, 2, 3))
+  ).rejects.toThrow('Failed to write long value');
+});
+
+it('writes short and long values using /card/write_short_and_long', async () => {
+  fetchMock.post(
+    '/card/write_short_and_long',
+    (url: string, mockRequest: MockRequest) => {
+      expect(url).toEqual('/card/write_short_and_long');
+      expect(mockRequest.headers).toEqual({
+        Accept: 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+      });
+      const body = mockRequest.body as string;
+      const values = body.split('&');
+      expect(values.length).toEqual(2);
+      expect(
+        values[0]?.startsWith('short_value=') &&
+          values[1]?.startsWith('long_value=')
+      ).toEqual(true);
```

Comment:
I know the short value doesn't have to come before the long value, and this test will break if the string concatenation in `writeShortAndLongValues` ever changes that order, but it didn't feel worth the complexity to support both orderings in this test

Reply:
I probably would have just checked for an exact expected string in the body haha, so I think you've already gone above and beyond.